tools/clang/lib/SPIRV/EmitVisitor.cpp

```diff
@@ -450,11 +441,15 @@ bool EmitVisitor::visit(SpirvExecutionMode *inst) {
 }
 
 bool EmitVisitor::visit(SpirvString *inst) {
+  auto it = stringIdMap.find(inst->getString());
+  if (it != stringIdMap.end())
+    return true;
```

Comment:
If I understand what you said correctly, you are worried that the caller of this method creates `SpirvString` as a temporary variable in the stack, right?

In my opinion, it might be fine because we do not access the `SpirvString` object after this method (we only access its result id after this method, not the object).
However, I agree that it looks hackey and unsafe.

I replace those temporary stock variables to pointers allocated by `new (SpirvContext) ...` and keep them in `spvInstructions` that will be destroyed in the destructor i.e., `EmitVisitor::~EmitVisitor()`.

Reply:
No I actually didn't mean that. Temporary stack variable is actually fine (preferred) in this case. Let's meet so I can explain my original comment.

Reply:
I updated `EmitVisitor::getOrAssignResultId<SpirvInstruction>()`